lib/adapter.js

```diff
@@ -155,6 +163,13 @@ class MochaAdapter {
         if (err) {
             this.runner.lastError = err
         }
+
+        // When starting a new test, propagate the details to the test runner
+        // so that commands, results, screenshots and hooks can be associated
+        // with this test
+        if (event === 'test:start') {
+            process.emit(event, message)
+        }
```

Comment:
ok I see there is no way around this. Limiting it to `test:start` sounds appropriate then